app/controllers/auth_key_pair_cloud_controller.rb

```diff
@@ -112,21 +112,16 @@ def create
       ext_management_system = find_by_id_filtered(ManageIQ::Providers::CloudManager, options[:ems_id])
       kls = kls.class_by_ems(ext_management_system)
       if kls.is_available?(:create_key_pair, ext_management_system, options)
-        begin
-          kls.create_key_pair(ext_management_system, options)
-          add_flash(_("Creating %{model} %{name}") % {
-            :model => ui_lookup(:table => 'auth_key_pair_cloud'),
-            :name  => options[:name]})
-        rescue => ex
-          add_flash(_("Unable to create %{model} %{name}. %{error}") % {
-            :model => ui_lookup(:table => 'auth_key_pair_cloud'),
-            :name  => options[:name],
-            :error => get_error_message_from_fog(ex.to_s)}, :error)
+        task_id = kls.create_key_pair_queue(session[:userid], ext_management_system, options)
+
+        add_flash(_("Key Pair creation failed: Task start failed: ID [%{id}]") %
+                  {:id => task_id.to_s}, :error) unless task_id.kind_of?(Integer)
```

Comment:
This is the pattern that's used in most existing code like this, but I see the problem. Looking at the code that generates the return value, the only way task_id won't be an integer is if something fails in task generation and it's somehow nil (rather than raising an exception). It seems that the right answer here would be to remove the ID from the flash message, since we won't have an ID in the failure case here.